lib/Sema/CSSolver.cpp

```diff
@@ -1557,7 +1601,7 @@ Constraint *ConstraintSystem::selectApplyDisjunction() {
 
     auto *applicable = &constraint;
     if (haveTypeInformationForAllArguments(
-            applicable->getFirstType()->castTo<AnyFunctionType>(), *this)) {
+            applicable->getFirstType()->castTo<AnyFunctionType>())) {
```

Comment:
It will always be a FunctionType, never a GenericFunctionType

Reply:
Will fix it. I seem to reflexively always use `AnyFunctionType` (and see it all over the place where it probably doesn't belong as well).